### PR TITLE
regression: path to --pdf-engine works again

### LINKED ISSUE #4681: --pdf-engine full path of the engine not working
According to the documentation:

`--pdf-engine=pdflatex|lualatex|xelatex|wkhtmltopdf|weasyprint|prince|context|pdfroff`

If the engine is not in your PATH, the full path of the engine may be specified here.
```
pandoc README.md -o out.pdf --pdf-engine=/usr/bin/xelatex
pdf-engine /usr/bin/xelatex not known
```
without full path it works:

```
pandoc README.md -o out.pdf --pdf-engine=xelatex
```
pandoc version used:
```
pandoc -v
pandoc 2.2.1
Compiled with pandoc-types 1.17.4.2, texmath 0.11, skylighting 0.7.0.2
```
It was working (with and without full path) in version 1.19.1
`/usr/bin/xelatex` exists and is executable.

### PR DESCRIPTION
**Pull request by mb21**
fixes #4681

### CODE REVIEW

File: src/Text/Pandoc/App.hs [hunk 1]
```diff
@@ -155,7 +155,7 @@ pdfWriterAndProg mWriter mEngine = do
     where
       go Nothing Nothing       = Right ("latex", "pdflatex")
       go (Just writer) Nothing = (writer,) <$> engineForWriter writer
-      go Nothing (Just engine) = (,engine) <$> writerForEngine engine
+      go Nothing (Just engine) = (,engine) <$> writerForEngine (takeBaseName engine)
       go (Just writer) (Just engine) =
            case find (== (baseWriterName writer, engine)) engines of
```
Comment (jgm) [inline src/Text/Pandoc/App.hs]: I think we also need
```
find (== (baseWriterName writer, takeBaseName engine)) engines
```
here, right?

Comment (mb21) [reply src/Text/Pandoc/App.hs]: good point, I updated the pull...

**Comment by jgm** [comment]
I also noticed that makePDF (in Text.Pandoc.PDF) pattern matches
on the engine passed in.  So, if an engine with a full path
is passed in, it won't match.  This probably needs to be fixed
too.  (You wouldn't notice it for latex, since the fallthrough
case does take the base name, but you should see an issue with
wkhtmltopdf or another program if a full path is specified.)

**Comment by mb21** [comment]
Indeed, I tested only with `latex`... rebased and hopefully we've caught them all now?

**Comment by jgm** [comment]
Thanks.  Some nice types would probably make this more solid, but this is good for now!

**Pull request merged**